From 2df8d324f91a1b90d57f5fe414ea9a0a8ca6ade2 Mon Sep 17 00:00:00 2001
From: Philipp Ammann <philipp.ammann@posteo.de>
Date: Tue, 11 Aug 2020 13:17:27 +0200
Subject: [PATCH 7/8] Guard X related #include's behind #ifdef MOZ_X11

The following headers are only needed on systems runnig X11:

  #include <gdk/gdkx.h>
  #include <gtk/gtkx.h>
  #include <X11/Xlib.h>
  #include <X11/Xutil.h>
---
 gfx/layers/client/TextureClient.cpp           |  5 +++--
 .../broker/SandboxBrokerPolicyFactory.cpp     |  4 +++-
 widget/gtk/mozwayland/mozwayland.c            |  4 +++-
 widget/gtk/nsClipboard.cpp                    | 20 +++++++++++++------
 widget/gtk/nsIdleServiceGTK.h                 |  9 ++++++---
 widget/gtk/nsWindow.cpp                       |  2 +-
 6 files changed, 30 insertions(+), 14 deletions(-)

diff --git a/gfx/layers/client/TextureClient.cpp b/gfx/layers/client/TextureClient.cpp
index d55de5a56cb4..92121c12e68e 100644
--- a/gfx/layers/client/TextureClient.cpp
+++ b/gfx/layers/client/TextureClient.cpp
@@ -55,8 +55,9 @@
 #  include "mozilla/layers/TextureClientX11.h"
 #endif
 #ifdef MOZ_WAYLAND
-#  include <gtk/gtkx.h>
-
+#  ifdef MOZ_X11
+#    include <gtk/gtkx.h>
+#  endif
 #  include "gfxPlatformGtk.h"
 #  include "mozilla/layers/WaylandDMABUFTextureClientOGL.h"
 #  include "mozilla/widget/nsWaylandDisplay.h"
diff --git a/security/sandbox/linux/broker/SandboxBrokerPolicyFactory.cpp b/security/sandbox/linux/broker/SandboxBrokerPolicyFactory.cpp
index 91e1b1cfc10a..99100e47ac47 100644
--- a/security/sandbox/linux/broker/SandboxBrokerPolicyFactory.cpp
+++ b/security/sandbox/linux/broker/SandboxBrokerPolicyFactory.cpp
@@ -38,7 +38,9 @@
 #  include <glib.h>
 #  ifdef MOZ_WAYLAND
 #    include <gdk/gdk.h>
-#    include <gdk/gdkx.h>
+#    ifdef MOZ_X11
+#      include <gdk/gdkx.h>
+#    endif
 #  endif
 #endif
 
diff --git a/widget/gtk/mozwayland/mozwayland.c b/widget/gtk/mozwayland/mozwayland.c
index 5cea1c5d1653..6e27d6366fc4 100644
--- a/widget/gtk/mozwayland/mozwayland.c
+++ b/widget/gtk/mozwayland/mozwayland.c
@@ -8,7 +8,9 @@
 #include <stdlib.h>
 #include "mozilla/Types.h"
 #include <gtk/gtk.h>
-#include <gtk/gtkx.h>
+#ifdef MOZ_X11
+#  include <gtk/gtkx.h>
+#endif
 #include <gdk/gdkwayland.h>
 
 union wl_argument;
diff --git a/widget/gtk/nsClipboard.cpp b/widget/gtk/nsClipboard.cpp
index 9eb3f3e0f9f6..fb4a10418c55 100644
--- a/widget/gtk/nsClipboard.cpp
+++ b/widget/gtk/nsClipboard.cpp
@@ -9,10 +9,6 @@
 
 #include "nsArrayUtils.h"
 #include "nsClipboard.h"
-#include "nsClipboardX11.h"
-#if defined(MOZ_WAYLAND)
-#  include "nsClipboardWayland.h"
-#endif
 #include "nsContentUtils.h"
 #include "HeadlessClipboard.h"
 #include "nsSupportsPrimitives.h"
@@ -31,7 +27,15 @@
 #include "imgIContainer.h"
 
 #include <gtk/gtk.h>
-#include <gtk/gtkx.h>
+
+#if defined(MOZ_X11)
+#  include "nsClipboardX11.h"
+#  include <gtk/gtkx.h>
+#endif
+
+#if defined(MOZ_WAYLAND)
+#  include "nsClipboardWayland.h"
+#endif
 
 #include "mozilla/Encoding.h"
 
@@ -91,12 +95,16 @@ nsClipboard::~nsClipboard() {
 NS_IMPL_ISUPPORTS(nsClipboard, nsIClipboard, nsIObserver)
 
 nsresult nsClipboard::Init(void) {
+#if defined(MOZ_X11)
   if (gfxPlatformGtk::GetPlatform()->IsX11Display()) {
     mContext = MakeUnique<nsRetrievalContextX11>();
 #if defined(MOZ_WAYLAND)
   } else {
     mContext = MakeUnique<nsRetrievalContextWayland>();
-#endif
+#endif /* MOZ_WAYLAND */
+#else
+  mContext = MakeUnique<nsRetrievalContextWayland>();
+#endif /* MOZ_X11 */
   }
   NS_ASSERTION(mContext, "Missing nsRetrievalContext for nsClipboard!");
 
diff --git a/widget/gtk/nsIdleServiceGTK.h b/widget/gtk/nsIdleServiceGTK.h
index d04aee3175af..e9610fd1f5ae 100644
--- a/widget/gtk/nsIdleServiceGTK.h
+++ b/widget/gtk/nsIdleServiceGTK.h
@@ -9,9 +9,12 @@
 #define nsIdleServiceGTK_h__
 
 #include "nsIdleService.h"
-#include <X11/Xlib.h>
-#include <X11/Xutil.h>
-#include <gdk/gdkx.h>
+
+#ifdef MOZ_X11
+#  include <X11/Xlib.h>
+#  include <X11/Xutil.h>
+#  include <gdk/gdkx.h>
+#endif
 
 typedef struct {
   Window window;               // Screen saver window
diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index 508dd49826c4..1ba847111014 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -45,7 +45,6 @@
 #include "ScreenHelperGTK.h"
 
 #include <gtk/gtk.h>
-#include <gtk/gtkx.h>
 
 #ifdef MOZ_WAYLAND
 #  include <gdk/gdkwayland.h>
@@ -53,6 +52,7 @@
 
 #ifdef MOZ_X11
 #  include <gdk/gdkx.h>
+#  include <gtk/gtkx.h>
 #  include <X11/Xatom.h>
 #  include <X11/extensions/XShm.h>
 #  include <X11/extensions/shape.h>
-- 
2.27.0
